In [77]:
import netCDF4 as nc  # library needed to read nc files
import numpy as np
import pandas as pd
import seaborn as sn

### Read data

In [78]:
# general variable data
ds = nc.Dataset("/Users/hauke/Ironhack/GitHub/Labs/Week_5_Midterm_Project/data/ERA5_SFC_MM_2021_10_FLOAT.nc")

# data for sea ice in september
ds_sit_sep = nc.Dataset("/Users/hauke/Ironhack/GitHub/Labs/Week_5_Midterm_Project/data/ERA5_SIC_202101-202110.nc")

# NAO index data
ds_svd = nc.Dataset("/Users/hauke/Ironhack/GitHub/Labs/Week_5_Midterm_Project/data/SVDcov_DJF_ERA5_1981-2021.nc")

In [79]:
# print information stores in data
#print(ds.__dict__)

In [81]:
# print dimensions
#for dim in ds_svd.dimensions.values():
#    print(dim)

In [41]:
# print variable metadata
#for var in ds_svd.variables.values():
#    print(var)

In [74]:
# print information for specific variable 
#print(ds_sit_sep['sic'])
#print(ds_svd["U"])

<class 'netCDF4._netCDF4.Variable'>
float32 sic(time, lat, lon)
    standard_name: sea_ice_cover
    long_name: Sea Ice Cover
    table: 128
    _FillValue: 1e+20
    missing_value: 1e+20
    units: 1
    cell_methods: time: mean
unlimited dimensions: time
current shape = (10, 192, 384)
filling on
<class 'netCDF4._netCDF4.Variable'>
float32 U(Zeit, Zeit)
unlimited dimensions: 
current shape = (41, 41)
filling on, default _FillValue of 9.969209968386869e+36 used


In [53]:
U = ds_svd["U"][:]

In [69]:
NAO = U[0,:]
NAO

masked_array(data=[ 0.09676873, -0.15498684,  0.17677438,  0.06572664,
                   -0.2245285 , -0.15692343, -0.12797922, -0.08410324,
                    0.28877774,  0.07248249,  0.03740686,  0.1442153 ,
                    0.18844478,  0.00854938,  0.15117285, -0.2730581 ,
                   -0.00728047, -0.07288415,  0.10884497,  0.22027102,
                   -0.13886958,  0.02792168, -0.12367234, -0.09500406,
                    0.0700616 , -0.13789874,  0.09778703,  0.10076685,
                   -0.08800685, -0.48961502, -0.22774799,  0.12997854,
                   -0.16169934,  0.01170531,  0.19052863,  0.09840314,
                    0.05412271,  0.02166592, -0.00554919,  0.20742963,
                   -0.20505777],
             mask=False,
       fill_value=1e+20,
            dtype=float32)

The dimensions are time (1 or 10 in the case of the additional sea ice file), lat (192) and lon (384).

In [76]:
# access data values
sst = ds["tos"][0,:,:]       # sea surface temperature
sit_oct = ds["sic"][0,:,:]   # october sea ice 
snd = ds["snd"][0,:,:]       # snow depth
ta = ds["ta100"][0,:,:]      # stratospheric temperature in 100 hPa
                                    # choose only september (=month 9) from months january(=1) - october(=10)
sit_sep = ds_sit_sep["sic"][8,:,:]  # september sea ice (taken from other file, see above) 

### Calculate mean of September and October sea ice variable

In [21]:
sit = (sit_sep + sit_oct) / 2

In [33]:
sit

masked_array(
  data=[[0.9228420257568359, 0.9225062131881714, 0.9221850633621216, ...,
         0.9244954586029053, 0.9239252805709839, 0.923357367515564],
        [0.9323089122772217, 0.9322522878646851, 0.932232141494751, ...,
         0.933390200138092, 0.932995080947876, 0.9326270222663879],
        [0.9398438930511475, 0.9407315850257874, 0.9416117072105408, ...,
         0.9373787641525269, 0.9381150603294373, 0.9389495849609375],
        ...,
        [--, --, --, ..., --, --, --],
        [--, --, --, ..., --, --, --],
        [--, --, --, ..., --, --, --]],
  mask=[[False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False],
        ...,
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True],
        [ True,  True,  True, ...,  True,  True,  True]],
  fill_value=1e+20,
  dtype=float32)

In [35]:
df = pd.DataFrame(sit)

In [36]:
df

0         1         2         3         4         5         6    \
0    0.922842  0.922506  0.922185  0.921864  0.921544  0.921220  0.920893   
1    0.932309  0.932252  0.932232  0.932123  0.931854  0.931546  0.931237   
2    0.939844  0.940732  0.941612  0.942443  0.943265  0.944068  0.944033   
3    0.933959  0.932903  0.931971  0.931003  0.930458  0.931285  0.932307   
4    0.939280  0.939555  0.939658  0.939522  0.939829  0.941349  0.943012   
..        ...       ...       ...       ...       ...       ...       ...   
187       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
188       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
189       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
190       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
191       NaN       NaN       NaN       NaN       NaN       NaN       NaN   

          7         8         9    ...       374       375       376  \
0    0.920558  0.920221  0.919868  ...  0.928688  0.927985  0.927316   
1    0.930960  0.930700  0.930318  ...  0.937889  0.937067  0.936186   
2    0.943648  0.943329  0.943015  ...  0.937443  0.936902  0.936354   
3    0.933264  0.934157  0.935538  ...  0.949738  0.949192  0.948165   
4    0.944502  0.945834  0.947492  ...  0.953544  0.952852  0.952245   
..        ...       ...       ...  ...       ...       ...       ...   
187       NaN       NaN       NaN  ...       NaN       NaN       NaN   
188       NaN       NaN       NaN  ...       NaN       NaN       NaN   
189       NaN       NaN       NaN  ...       NaN       NaN       NaN   
190       NaN       NaN       NaN  ...       NaN       NaN       NaN   
191       NaN       NaN       NaN  ...       NaN       NaN       NaN   

          377       378       379       380       381       382       383  
0    0.926733  0.926164  0.925608  0.925049  0.924495  0.923925  0.923357  
1    0.935311  0.934640  0.934191  0.933787  0.933390  0.932995  0.932627  
2    0.935887  0.935616  0.935826  0.936647  0.937379  0.938115  0.938950  
3    0.946337  0.944337  0.942351  0.941118  0.939760  0.937826  0.935809  
4    0.951322  0.949508  0.947404  0.945565  0.943959  0.942109  0.940348  
..        ...       ...       ...       ...       ...       ...       ...  
187       NaN       NaN       NaN       NaN       NaN       NaN       NaN  
188       NaN       NaN       NaN       NaN       NaN       NaN       NaN  
189       NaN       NaN       NaN       NaN       NaN       NaN       NaN  
190       NaN       NaN       NaN       NaN       NaN       NaN       NaN  
191       NaN       NaN       NaN       NaN       NaN       NaN       NaN  

[192 rows x 384 columns]